lib/graphql/execution/execute.rb

```diff
@@ -164,14 +162,19 @@ def resolve_value(parent_type, field_defn, field_type, value, selection, field_c
             result
           when GraphQL::TypeKinds::NON_NULL
             wrapped_type = field_type.of_type
-            resolve_value(
+            inner_value = resolve_value(
               parent_type,
               field_defn,
               wrapped_type,
               value,
               selection,
               field_ctx,
             )
+            if inner_value.nil?
+              PROPAGATE_NULL
```

Comment:
This can happen for non-null union or interface types. The non-null call to `resolve` passes the value to the union or interface type, but then the object fails type resolution, so it gets replaced with `nil`.  Here's the [spec](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/compatibility/execution_specification.rb#L357). I added some [similar code](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/query/serial_execution/value_resolution.rb#L53-L57) to make SerialExecution pass and must have assumed it was required here too! 

But I think this is handled by `FieldResult` in this case. Removed in 4b02c42. Thanks!